include/aspect/particle/property/interface.h

```diff
@@ -0,0 +1,355 @@
+           * saves considerable computation time in cases, when no plugin needs
+           * to update tracer properties over time.
+           */
+          virtual
+          UpdateTimeFlags
+          need_update ();
+
+          virtual
+          void
+          data_length(std::vector<unsigned int> &length) const = 0;
+
+          /**
+           * Set up the name information for the particle property
+           *
+           * @param [in,out] names Vector that contains the property name
+           */
+          virtual
+          void
+          data_names(std::vector<std::string> &names) const = 0;
+
+
+          /**
+           * Declare the parameters this class takes through input files.
+           * Derived classes should overload this function if they actually do
+           * take parameters; this class declares a fall-back function that
+           * does nothing, so that property classes that do not take any
+           * parameters do not have to do anything at all.
+           *
+           * This function is static (and needs to be static in derived
+           * classes) so that it can be called without creating actual objects
+           * (because declaring parameters happens before we read the input
+           * file and thus at a time when we don't even know yet which
+           * property objects we need).
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           * The default implementation in this class does nothing, so that
+           * derived classes that do not need any parameters do not need to
+           * implement it.
+           */
+          virtual
+          void
+          parse_parameters (ParameterHandler &prm);
+      };
+
+
+
+      /**
+       * Manager class of properties - This class sets the data of the particles
+       * and updates it over time if requested by the user selected properties
+       */
+      template <int dim>
+      class Manager : public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Empty constructor for Manager
+           */
+          Manager ();
+
+          /**
+           * Destructor for Manager
+           */
+          virtual
+          ~Manager ();
+
+          /**
+           * Initialization function. This function is called once at the
+           * beginning of the program after parse_parameters is run.
+           */
+          virtual
+          void
+          initialize ();
+
+          /**
+           * Initialization function for particle properties. This function is
+           * called once at the creation of a particle
+           */
+          virtual
+          void
+          initialize_particle (Particle<dim> &particle,
+                               const Vector<double> &solution,
+                               const std::vector<Tensor<1,dim> > &gradients);
```

Comment:
You are right, something should make clear that this object just does something to a different object (the particle) and nothing to itself. I renamed the functions in the manager to `initialize_one_particle` and `update_one_particle`, and the ones in the individual plugins to `initialize_one_particle_property`, and `update_one_particle_property`. Additionally, I made all of these functions `const`, because they really should not change anything in `this`.